### PR TITLE
add instructions and meta to data model diagrams

### PR DESCRIPTION
**Pull request by fainashalts**

### CODE REVIEW

File: docs/data-model.rst [hunk 1]
```diff
@@ -96,6 +97,7 @@ Bytecode is represented by an object that contains:
 * An unlinked bytes array (``bytes``)
 * A list of any known link references (``linkReferences``)
 * Possibly a mapping to known source ranges (``sourceMap``)
+* An instructions object containing information derived from the bytecode and sourcemap (``instructions``)
```
Comment (fainashalts) [inline docs/data-model.rst]: Not sure if I'm really describing this properly. Happy to reword!

File: docs/data-model.rst [hunk 2]
```diff
@@ -123,7 +126,23 @@ Each link reference is represented by an object that specifies:
...
+   }
+
+  object(Meta) {
+    + cost: Integer
+    + pops: Integer
+    + pushes: Integer
+    + dynamic: Integer
+   }   
 
+   Bytecode *-- "1" Instructions
```
Comment (gnidan) [inline docs/data-model.rst]: So I did the "collection resource" thing for `Sources`, for instance, but I'm backpedaling on that now. I think we should get rid of that and instead only have direct `"n"` relations. So in this situation, I'd expect this to be either:

```
Bytecode *-- "1" Instructions
Instructions *-- "n" Instruction
```

or what I now prefer:
```
Bytecode *-- "n" Instruction
```

What do you think?

Comment (fainashalts) [reply docs/data-model.rst]: I think that makes sense. Changed it to the second way:
 
`Bytecode *-- "n" Instruction`

File: docs/data-model.rst [hunk 3]
```diff
@@ -54,6 +54,7 @@ This data model distinguishes between |ContractTypes| and |ContractInstances|:
 
    object(Bytecode) {
     + bytes: Array<Byte>
+    + instructions: Instruction
```
Comment (gnidan) [inline docs/data-model.rst]: should be `Array<Instruction>`

File: docs/data-model.rst [hunk 4]
```diff
@@ -110,6 +112,7 @@ Each link reference is represented by an object that specifies:
    object(Bytecode) {
     + bytes: Array<Byte>
     + sourceMap : SourceMap
+    + instructions: Instruction
```
Comment (gnidan) [inline docs/data-model.rst]: (Same as above)

File: docs/data-model.rst [hunk 5]
```diff
@@ -190,9 +209,25 @@ of |SourceMaps|.
...
```
Comment (gnidan) [inline docs/data-model.rst]: (same as above)

File: docs/data-model.rst [hunk 6]
```diff
@@ -96,6 +97,7 @@ Bytecode is represented by an object that contains:
 * An unlinked bytes array (``bytes``)
 * A list of any known link references (``linkReferences``)
 * Possibly a mapping to known source ranges (``sourceMap``)
+* An instruction object containing information derived from the bytecode and sourcemap (``instruction``)
```
Comment (gnidan) [inline docs/data-model.rst]: This should say something like `An array of instruction objects containing...`, and have the corresponding correct field `instructions`

**Pull request merged**